internal/venice-common/src/main/java/com/linkedin/venice/helix/MapKeyStringAnnotatedReadOnlySchemaRepository.java

```diff
@@ -0,0 +1,176 @@
+package com.linkedin.venice.helix;
+
+import static com.linkedin.venice.schema.SchemaUtils.getAnnotatedStringMapDerivedSchemaEntry;
+import static com.linkedin.venice.schema.SchemaUtils.getAnnotatedStringMapValueSchemaEntry;
+
+import com.linkedin.venice.meta.ReadOnlySchemaRepository;
+import com.linkedin.venice.schema.SchemaEntry;
+import com.linkedin.venice.schema.rmd.RmdSchemaEntry;
+import com.linkedin.venice.schema.writecompute.DerivedSchemaEntry;
+import com.linkedin.venice.utils.Pair;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * This class is a wrapper schema repository class, which is only used by merge conflict resolver in A/A store ingestion
+ * task. This class will annotate value schema and partial update schema so when performing Map collection merging,
+ * Map field's key can be deserialized into String type and thus is comparable for DCR purpose. Without this wrapper or
+ * user annotation, map fields will have key in UTF-8 type which is not comparable.
+ * This wrapper class only implements retrieval of the value schema (including the latest value schema and superset schema)
+ * and partial update schema as they are the only usage in merge conflict resolver. Other operations are not supported
+ * intentionally to avoid unexpected behavior.
+ */
+public class MapKeyStringAnnotatedReadOnlySchemaRepository implements ReadOnlySchemaRepository {
```

Comment:
Although in general I think it's a good intuition to reuse existing interfaces, I think in this case we may benefit from a more specialized API. It would be nice to avoid doing a lookup by store name as part of all functions. Instead, I would prefer if we had an API which is for getting the schemas of a single store, and therefore skip the per-store lookups. We can pre-instantiate this object in the SIT and use it for a single store in that context.

Could we use the `SchemaReader` interface to achieve this?

Reply:
If `SchemaReader` doesn't work, then maybe just a new class...?

Reply:
SchemaReader does not work, like getSupersetOrLatestValueSchema would probably not do it. I pushed a new change to make it a unique class, but probably you are right, since it is only used in SIT, storeName is fixed. We don't need to perform additional look up then. I will change it again.